include/exadg/postprocessor/write_output.h

```diff
@@ -78,13 +81,60 @@ write_boundary_IDs(dealii::Triangulation<dim> const & triangulation,
 template<int dim>
 void
 write_grid(dealii::Triangulation<dim> const & triangulation,
+           dealii::Mapping<dim> const &       mapping,
+           unsigned int const                 n_subdivisions,
            std::string const &                folder,
-           std::string const &                file)
+           std::string const &                file,
+           unsigned int const &               counter,
+           MPI_Comm const &                   mpi_comm)
 {
-  std::string filename = folder + file + "_grid";
+  std::string filename = file + "_grid";
 
-  dealii::GridOut grid_out;
-  grid_out.write_mesh_per_processor_as_vtu(triangulation, filename);
+  dealii::DataOut<dim> data_out;
+
+  dealii::DataOutBase::VtkFlags flags;
+  flags.write_higher_order_cells = n_subdivisions > 1;
+  data_out.set_flags(flags);
+
+  data_out.attach_triangulation(triangulation);
+  data_out.build_patches(mapping, 4, dealii::DataOut<dim>::curved_inner_cells);
+  data_out.write_vtu_with_pvtu_record(folder, filename, counter, mpi_comm, 4);
+}
+
+template<int dim>
+void
+write_points(std::vector<dealii::Point<dim>> const & points,
+             std::string const &                     folder,
+             std::string const &                     file,
+             unsigned int const &                    counter,
+             MPI_Comm const &                        mpi_comm)
+{
+  std::string filename = file + "_points";
+
+  double min_coord = points[0][0];
+  double max_coord = points[0][0];
+  for(auto const & p : points)
+  {
+    for(unsigned int d = 0; d < dim; ++d)
+    {
+      min_coord = std::min(p[d], min_coord);
+      max_coord = std::max(p[d], max_coord);
+    }
+  }
+  min_coord = dealii::Utilities::MPI::min(min_coord, mpi_comm);
+  max_coord = dealii::Utilities::MPI::max(max_coord, mpi_comm);
```

Comment:
Let me add two comments to the code:
- I would compute the `min/max` for each coordinate, not all coordinates, and then `GridGenerator::hyper_rectangle`. The question is then if this should be somehow combined with `dealii::BoundingBox` - today, there is a function https://github.com/dealii/dealii/blob/d47537fa246886e5308d5aecbf42358d73a00774/include/deal.II/grid/grid_tools.h#L380-L382 but not the other way around. (Would that be useful?)
- If performance is important, one should compute `Utilities::MPI::max({-min_coord, max_coord}, mpi_comM);` in a single go (not sure of this syntax works, but you get what I mean). The reason is that global communication is expensive, so doing it once on 2 elements (or `2 * dim` elements when computing a hyper rectangle) lowers compute time by a factor of 2 in this particular function. I am not saying we should do that here, but the MPI person in my considers this the better programming style with MPI, so I should mention it.

Reply:
thanks for the hint! in the future if I can't avoid `Utilities::MPI::max` or the like, I will consider such a strategy!
the `BoundingBox` and `Hyperrectangle` removed some code here as well!